src/Integration.Vsix.UnitTests/Helpers/IDEWindowServiceTests.cs

```diff
@@ -0,0 +1,148 @@
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration.Vsix;
+using SonarLint.VisualStudio.Integration.Vsix.Native;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Helpers
+{
+    [TestClass]
+    public class IDEWindowServiceTests
+    {
+        private readonly IntPtr ValidHandle = new IntPtr(123);
+        private Mock<INativeMethods> nativeMock;
+        private TestLogger logger;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            nativeMock = new Mock<INativeMethods>();
+            logger = new TestLogger(logToConsole: true);
+        }
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<IDEWindowService, IIDEWindowService>(null,
+                new[] { MefTestHelpers.CreateExport<ILogger>(Mock.Of<ILogger>()) });
+        }
+
+        [TestMethod]
+        public void BringToFront_InvalidHandle_NoApiCalls()
+        {
+            var testSubject = new IDEWindowService(nativeMock.Object, IntPtr.Zero, logger);
+
+            // Act
+            testSubject.BringToFront();
+
+            nativeMock.Invocations.Count.Should().Be(0);
+        }
+
+        [TestMethod]
+        public void BringToFront_GetWindowPlacementFails_IsStillBroughtToFront()
+        {
+            var testSubject = new IDEWindowService(nativeMock.Object, ValidHandle, logger);
+
+            SetGetPlacementResponse(false, new WINDOWPLACEMENT());
+
+            // Act
+            testSubject.BringToFront();
+
+            CheckGetPlacementIsCalled();
+            CheckShowWindowIsNotCalled();
+            CheckSetForegroundIsCalled();
+        }
+
+        [TestMethod]
+        public void BringToFront_NotMinimized_IsBroughtToFrontButNotRestored()
+        {
+            var testSubject = new IDEWindowService(nativeMock.Object, ValidHandle, logger);
+
+            SetGetPlacementResponse(result: true,
+                new WINDOWPLACEMENT { showCmd = 0 });
```

Comment:
Can this be a readonly? 
`WINDOWPLACEMENT NotMinimizedPlacement` 
It would make the tests slightly easier to read (same for the MinimizedPlacement below)